File: fix_admin_urls.js
const fs = require('fs');
const path = require('path');

function fixFile(filepath) {
    const content = fs.readFileSync(filepath, 'utf8');
    
    // Pattern for src={`${process.env.NEXT_PUBLIC_API_URL}${var}`}
    // We replace it with: src={(var)?.startsWith('http') ? (var) : `${process.env.NEXT_PUBLIC_API_URL}${var}`}
    const pattern = /src=\{`\$\{process\.env\.NEXT_PUBLIC_API_URL\}\$\{([^}]+)\}`\}/g;
    const replacement = 'src={($1)?.startsWith("http") ? ($1) : `${process.env.NEXT_PUBLIC_API_URL}${$1}`}';
    
    const newContent = content.replace(pattern, replacement);
    
    if (newContent !== content) {
        fs.writeFileSync(filepath, newContent, 'utf8');
        console.log(`Fixed ${filepath}`);
    }
}

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walkDir(fullPath);
        } else if (fullPath.endsWith('.tsx')) {
            fixFile(fullPath);
        }
    }
}

const targetDir = path.join(__dirname, 'admin_frontend', 'src');
walkDir(targetDir);
console.log("Done");

File: fix_admin_urls.py
import os
import re

def fix_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Find and replace src={`${process.env.NEXT_PUBLIC_API_URL}${variable}`}
    pattern1 = r'src=\{`\$\{process\.env\.NEXT_PUBLIC_API_URL\}\$\{([^}]+)\}`\}'
    replacement1 = r'src={(\1)?.startsWith("http") ? (\1) : `${process.env.NEXT_PUBLIC_API_URL}${\1}`}'
    
    # Also handle some edge cases like `admin_frontend/src/app/landing/page.tsx` line 519:
    # src={feature.icon.startsWith('/') ? `${process.env.NEXT_PUBLIC_API_URL}${feature.icon}` : feature.icon}
    # This one is probably fine if it uses .startsWith('/'), but actually cloudinary URLs might not start with '/'
    
    new_content = re.sub(pattern1, replacement1, content)
    
    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Fixed {filepath}")

def main():
    directory = r"d:\work\Meewa\MeewaIndustries\admin_frontend\src"
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith('.tsx'):
                fix_file(os.path.join(root, file))

if __name__ == "__main__":
    main()
